packages/cubejs-server-core/src/core/server.ts

```diff
@@ -569,88 +569,49 @@ export class CubejsServerCore {
   public async getOrchestratorApi(context: RequestContext): Promise<OrchestratorApi> {
     const orchestratorId = await this.contextToOrchestratorId(context);
 
-    if (this.orchestratorStorage.has(orchestratorId)) {
-      return this.orchestratorStorage.get(orchestratorId);
-    }
-
-    /**
-     * Hash table to store promises which will be resolved with the
-     * datasource drivers. DriverFactoryByDataSource function is closure
-     * this constant.
-     */
-    const driverPromise: Record<string, Promise<BaseDriver>> = {};
-
-    let externalPreAggregationsDriverPromise: Promise<BaseDriver> | null = null;
-
-    const contextToDbType: DbTypeAsyncFn = this.contextToDbType.bind(this);
-    const externalDbType = this.contextToExternalDbType(context);
-
-    // orchestrator options can be empty, if user didn't define it.
-    // so we are adding default and configuring queues concurrency.
-    const orchestratorOptions =
```

Comment:
As you can see, there is an await point here at `(await this.orchestratorOptions(context)) || {},`. With this, it's possible to have two orchestrator API instances, if it's called immediately at one time.